package/MDAnalysis/topology/LAMMPSParser.py

```diff
@@ -188,97 +258,122 @@ def iterdata(self):
-        sections - dict of section name: content
+        header: dict of str
+            A dict in which a key and value pair is a 'header' name and
+            its value in str foramt.
+        sections: dict of list of str
+            A dict in which a key and value pair is a 'section' name and
+            a list of strings (relevenat lines of the data file).
         """
         f = list(self.iterdata())
 
+        # finding the starting line of each section:
         starts = [i for i, line in enumerate(f)
                   if line.split()[0] in SECTIONS]
+        # adding None to properly handle lines in the last SECTION below
         starts += [None]
 
         header = {}
+
+        # lines before the first SECTION found in the file are about HEADERS:
         for line in f[:starts[0]]:
             for token in HEADERS:
                 if line.endswith(token):
                     header[token] = line.split(token)[0]
                     continue
 
-        sects = {f[l]:f[l+1:starts[i+1]]
+        sects = {f[l]: f[l+1:starts[i+1]]  # f[l] is indeed a SECTION name
                  for i, l in enumerate(starts[:-1])}
 
         return header, sects
 
     @staticmethod
     def _interpret_atom_style(atom_style):
-        """Transform a string description of atom style into a dict
-
-        Required fields: id, type, x, y, z
-        Optional fields: resid, charge
-
-        eg: "id resid type charge x y z"
-        {'id': 0,
-         'resid': 1,
-         'type': 2,
-         'charge': 3,
-         'x': 4,
-         'y': 5,
-         'z': 6,
-        }
+        """Map field names in a given `atom_style` to their corresponding
+        locations.
+
+        Required attributes: id, type, x, y, z
+        Optional attributes: resid, charge
+        Note: other attributes are ignored.
+
+        Exammples
+        ---------
+        "full" atom_style has the following attributes:
+            "id resid type charge x y z"
+        `_interpret_atom_style` returns this dict for "full" atom_style:
+            {'id': 0,
+             'resid': 1,
+             'type': 2,
+             'charge': 3,
+             'x': 4,
+             'y': 5,
+             'z': 6,
+            }
+
+        Arguments
+        ---------
+        atom_style: str, {'angle', 'atomic', 'body', 'bond', 'charge',
+                          'dipole', 'dpd', 'edpd', 'electron', 'ellipsoid',
+                          'full', 'line', 'mdpd', 'mesont', 'molecular',
+                          'peri', 'smd', 'sph', 'sphere', 'spin', 'tdpd',
+                          'template', 'tri', 'wavepacket', 'hybrid'}
+            Name of one of the defined atom styles.
+
+        Returns
+        -------
+        dict:
+            a map between the attributes' names and attributes' locations.
         """
         style_dict = {}
 
-        atom_style = atom_style.split()
+        for loc, attr in enumerate(ATOM_STYLES[atom_style]):
```

Comment:
This can be a dictionary comprehension:

```python
style_dict = {attr:loc for loc,attr in enumerate(ATOM_STYLES[atom_style])}
```
or something like that.